### PR TITLE
Remote x11 forwarding

### PR DESCRIPTION
**Pull request by jaketockerman**
use --net=host on server and ssh -Y to pass x11 forwarding over ssh

### CODE REVIEW

**Review by jrbaxter0 — APPROVED** [review]
As long as you provided **detailed** instructions to members over Discord, this is fine with me. I have not reviewed the server functionality at all since that is not my domain. When this is merged, I expect there to be an @rocketleague announcement on #rl-general with the exact copy-paste command to rename the image.

**Comment by jrbaxter0** [comment]
But as a comment, if you use `--net host` when running on the server, only one person can be running ROS stuff at a time by default. It will use port 11311 and get upset that there is already something running. You can change the default port, which is something I do for training hundreds of simultaneous environments on the super computer. But that code is somewhat flakey.

Probably you want to bind only the port actually needed for display.

**Comment by jrbaxter0** [comment]
It also might be better to use this solution so that each user's docker images / containers are specific to themselves. Then you don't need to worry about people killing other's stuff, and doing this tagging stuff: https://forums.docker.com/t/share-docker-between-users/5728

**Comment by jrbaxter0** [comment]
But again, whatever solution you decide on for the server is fine as long as it doesn't break the repo.

File: docker/docker-run.sh [hunk 1]
```diff
@@ -20,8 +20,9 @@ docker run --rm \
     -v $WS_DIR:/home/$USER/catkin_ws \
     -v /tmp/.X11-unix:/tmp/.X11-unix \
     `[ -f ~/.gitconfig ] && echo "-v $HOME/.gitconfig:/home/$USER/.gitconfig:ro"` \
+    -v $HOME/.Xauthority:/home/$USER/.Xauthority \
```
Comment (jrbaxter0) [inline docker/docker-run.sh:23]: This is not portable to Mac. We already do this exact thing on line 19, but do it as read only, which I'd prefer unless if there is a need to be able to change it.

**Review by jrbaxter0 — CHANGES_REQUESTED** [review]
My review is primarily concerned with "will this break the project?" as opposed to "will this work on the server?"

That being said, my two major issues are with the comments in made in `docker-run.sh`. All the other comments are just suggestions.

I also think we should update the Docker README and link to it in the main README like we do other sections.

**Comment by hmccarty** [comment]
Works for me

**Review by jrbaxter0 — APPROVED** [review]
Changes to `-build`, `-join`, `-run` look good to me. Will not mess up any existing workflows if containers are rebuilt when merged.

**Pull request merged**